pkg/resource/table/conditions.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package table
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+)
+
+// getSyncedCondition returns the Condition in the resource's Conditions
+// collection that is of type ConditionTypeResourceSynced. If no such condition
+// is found, returns nil.
+//
+// TODO(jaypipes): Move to ACK code-gen templates.
```

Comment:
?? Are these files not auto generated?

Reply:
Not yet, I guess we still need to experiment with other controllers to figure out a general design

/cc @jaypipes

Reply:
@RedbackThomson yeah, eventually I'd like to generate functions like these and/or move into ACK runtime. I added these for the mq-controller:

https://github.com/aws-controllers-k8s/mq-controller/blob/main/pkg/resource/broker/conditions.go
https://github.com/aws-controllers-k8s/mq-controller/blob/main/pkg/resource/broker/hooks.go

and then adapted them for the RDS controller:

https://github.com/aws-controllers-k8s/rds-controller/blob/main/pkg/resource/db_instance/conditions.go
https://github.com/aws-controllers-k8s/rds-controller/blob/main/pkg/resource/db_instance/hooks.go

I use the service controllers as a bit of a laboratory for experimentation in this way while I figure out what works as a standard.

Reply:
> 3 controllers using same code, good time to move ?
> [aws-controllers-k8s/community#806](https://github.com/aws-controllers-k8s/community/issues/806)

after I get those logging changes pushed up, yeah...